### PR TITLE
Changes for v2.1

### PR DESCRIPTION
**Pull request by raoulbhatia**
* Support newer Ubuntu releasese: 17.10 and 18.04
* Update Ruby/Puppet test matrix
* Add virtio-rng
* Add domain::machine_type
* Define default disk bus in domain

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -25,6 +25,10 @@
           $libvirt_package_names  = ['libvirt-bin']
           $service_name           = 'libvirtd'
         }
+        'bionic': {
+          $libvirt_package_names  = ['libvirt-daemon-system', 'libvirt-clients']
+          $service_name           = 'libvirtd'
```
[the remark this answers is not in the document]
Comment (raoulbhatia) [reply manifests/params.pp]: `libvirt-bin` is marked as a transitional package that can safely be removed.  Actually, it was already marked this way in Artful.

I am a fan of following such changes and to not rely on transitional packages as might be removed eventually.

Having said that, I currently don't have strong feelings about that right now.  It just pops up when using `deborphan`.

[the remark this answers is not in the document]
Comment (raoulbhatia) [reply manifests/params.pp]: Please let me know if you'd like to keep `libvirt-bin` or if you'd like to follow the transition of dependencies to `libvirt-daemon-system` and `libvirt-clients`.

File: manifests/params.pp [hunk 2]
```diff
@@ -21,6 +21,10 @@
           $libvirt_package_names  = ['libvirt-bin', 'qemu-kvm']
           $service_name           = 'libvirt-bin'
         }
+        'artful': {
+          $libvirt_package_names  = ['libvirt-bin']
+          $service_name           = 'libvirtd'
```
[the remark this answers is not in the document]
Comment (raoulbhatia) [reply manifests/params.pp]: Dropped by mistake as I didn't realize the transition from `qemu` to `qemu-kvm`.  Will fix that.

File: manifests/params.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (raoulbhatia) [reply manifests/params.pp]: Last remark in this regard - I see that stretch has already dropped `libvirt-bin` in favor of `libvirt-daemon-system`.  How about we just align Ubuntu 18.04 LTS to that?

**Comment by raoulbhatia** [comment]
Quick coding style question:  Do you prefer me to address your comments by adding additional commits on top, or shall I amend the relevant commits instead?

Thanks,
Raoul

File: manifests/params.pp (hunk 1 above)
[the remark this answers is not in the document]
Comment (trefzer) [reply manifests/params.pp]: oh I did not realize that !
It's probably fine to align with debian. At least if they are using the same source (which i did not check for libvirt...)

File: manifests/params.pp (hunk 2 above)
[the remark this answers is not in the document]
Comment (trefzer) [reply manifests/params.pp]: ok. same in bionic !

**Comment by trefzer** [comment]
I better like amendments. So we have at the end one propper commit (or several if we add a combined function).
For the issues I like one issue per subject. So I can merge independently. (eg. Update Test Matrix is a great commit, which I could not merge easily because the others had discussion points.

But anyway, I appreciate you're work, so as soon as the the discussion points are done, I will merge.

File: manifests/params.pp (hunk 2 above)
[the remark this answers is not in the document]
Comment (raoulbhatia) [reply manifests/params.pp]: Fixed in #19

**Pull request closed without merging**